Feature (Helm): Add library chart

hickeyma opened the pull request:
**What this PR does / why we need it**:
This PR differentiate between an application chart and a library chart by defining a new `type` field in `Chart.yaml` . Library charts can be used as dependencies but are not installable.

Closes  #5085
partial #5397

*Special notes for your reviewer**:
Update on design proposal: https://github.com/helm/helm/issues/5085#issuecomment-462385310

*If applicable**:
- [x] this PR contains documentation
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

hickeyma committed:
Add chart type

The chart type is added to differentiaite between an application chart
and a library chart. Library charts can be used as dependencies but are
not installable.

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma committed:
Update the docs

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

adamreese commented:
/cc @mattfarina @prydonius

technosophos commented on cmd/helm/install.go:
```diff
@@ -231,6 +231,14 @@ func (o *installOptions) run(out io.Writer) error {
 		return err
 	}
 
+	chartType := chartRequested.Metadata.Type
```
I see these same tests multiple times in this PR. Can we just make a function?

hickeyma replied:
Good idea. Function added.

technosophos reviewed (approved):
I'm good with this. I left a minor nit comment, but all is looking good for adding this as a type.

hickeyma committed:
Update after review

Review comment:
- https://github.com/helm/helm/pull/5295#pullrequestreview-203519813

Signed-off-by: Martin Hickey <martin.hickey@ie.ibm.com>

hickeyma commented:
Thanks @technosophos for the review and comment. 

Review comment updated and ready for review again.

hickeyma commented:
Thanks @adamreese for merging.
